pupil_src/shared_modules/display_recent_gaze.py

```diff
@@ -25,14 +27,23 @@ def __init__(self, g_pool):
         self.pupil_display_list = []
 
     def recent_events(self, events):
+        frame = events.get("frame", None)
+        if not frame:
+            return
+        frame_size = frame.width, frame.height
```

Comment:
It might make sense to cache this variable and to use it once `frame` is not avilable.